src/main/webapp/2Dscript.js

```diff
@@ -414,42 +489,73 @@ function placeRectangleEnd() {
   if (infoWindow !== undefined) {
     infoWindow.setMap(null);
   }
-  // Clear other paths.
-  // line
-  // minLatPoint
-  // lat , lng
-  // minLngPoint
-  // lat, lng
-  const subSectionData = computeSubSection(currentRun.data,
-      currentLat, priorLat, currentLng, priorLng);
-  // Choose a subSectionNumber, implement differently in future pr.
-  const subSectionNumber = 1;
-  // Clear prior paths, only display newly selected ones.
-  if (markerBottom !== undefined) {
-    markerBottom.setMap(null);
+  clearSelectedPaths(['subSection', 'markerBottom', 'markerTop']);
+  // Clear other paths. line minLatPoint lat , lng minLngPoint lat, lng Iterate
+  // over all data values here, need to find a way to check the checkbox value,
+  // should likely store this in an object somewhere to prevent excess DOM
+  // queries.
+  const mapRuns = Object.entries(runs);
+  for ([id, currentRun] of mapRuns) {
+    if (!currentRun.checkBox.checked) {
+      continue;
+    }
+    const subSectionData = computeSubSection(currentRun.data,
+        currentLat, priorLat, currentLng, priorLng);
+    currentRun.subData = subSectionData.subData;
+    // Clear prior paths, only display newly selected ones.
+    const subPath = getPolyLine(currentRun.subData, 'blue', 1.0, 2, 1000);
+    // Setup event listener to show option for 3D window when polyline is
+    // clicked.
+    currentRun.subSection = subPath;
+    subPath.setMap(map);
+    currentRun.markerBottom = genMarker(subSectionData.minLatPoint.lat,
+        subSectionData.minLatPoint.lng);
+    currentRun.markerTop = genMarker(subSectionData.maxLngPoint.lat,
+        subSectionData.maxLngPoint.lng);
+    showAll([currentRun.subSection, currentRun.markerBottom,
+      currentRun.markerTop]);
   }
-  if (markerTop !== undefined) {
-    markerTop.setMap(null);
+  const subSectionObject = {};
+  for ([id, currentRun] of mapRuns) {
+    if (!(currentRun.checkBox.checked && currentRun.subData.length !== 0)) {
+      continue;
+    }
+    subSectionObject[id] = {};
+    subSectionObject[id].color = currentRun.color.value;
+    subSectionObject[id].data = currentRun.subData;
+    console.log(subSectionObject);
+    google.maps.event.addListener(currentRun.subSection,
+        'click', (event) => createInfoWindow(event, subSectionObject));
```

Comment:
Why loop through twice instead of doing it all together? (It may help to pull each of the current loop bodies into a helper function if you do collapse them.)